source/winVersion.py

```diff
@@ -34,21 +78,43 @@ def __init__(
 		self.major = major
 		self.minor = minor
 		self.build = build
-		self.releaseName = releaseName
+		if releaseName:
+			self.releaseName = releaseName
+		else:
+			self.releaseName = self._getWindowsReleaseName()
 		self.servicePack = servicePack
 		self.productType = productType
 
-	def __repr__(self):
-		winVersionText = []
-		if self.releaseName:
-			winVersionText.append(self.releaseName)
+	def _getWindowsReleaseName(self) -> str:
+		"""Returns the public release name for a given Windows release based on major, minor, and build.
+		This is useful if release names are not defined when constructing this class.
+		For example, 6.1 will return 'Windows 7'.
+		For Windows 10, feature update release name will be included.
+		On server systems, unless noted otherwise, client release names will be returned.
+		For example, 'Windows 10 1809' will be returned on Server 2019 systems.
+		"""
+		if (self.major, self.minor) == (6, 1):
+			return "Windows 7"
+		elif (self.major, self.minor) == (6, 2):
+			return "Windows 8"
+		elif (self.major, self.minor) == (6, 3):
+			return "Windows 8.1"
+		elif self.major == 10:
+			# From Version 1511 (build 10586), release Id/display version comes from Windows Registry.
+			# However there are builds with no release name (Version 1507/10240)
+			# or releases with different builds.
+			# Look these up first before asking Windows Registry.
+			if self.build in _BUILDS_TO_RELEASE_NAMES:
+				return _BUILDS_TO_RELEASE_NAMES[self.build]
+			releaseId = _getRunningVersionNameFromWinReg()
+			if not releaseId:
+				releaseId = "unknown"

```

Comment:
`_getWindowsReleaseName` shouldn't fall back to the registry, the only case where the registry should be used is when getting the version for the version currently running.

```suggestion
			if self.build in _BUILDS_TO_RELEASE_NAMES:
				return _BUILDS_TO_RELEASE_NAMES[self.build]
			if not releaseId:
				releaseId = "unknown"

```